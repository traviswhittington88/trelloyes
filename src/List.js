import React from 'react';
import Card from './Card.js';



function List(props) {
const CardComponent = [];

for(let i = 0; i < props.cardsArray.length; i++) {
  for(let j =0; j < props.cardsObj.length; j++) {
    if(props.cardsArray[i] === props.cardsObj[j].id){
      CardComponent.push(<Card title={props.cardsObj[j].title}
        content = {props.cardsObj[j].content}
        key = {props.cardsObj[j].id}/>)
        i = i+1;
    }
    else{
      j=+j;
    }
    
  }
}

  return (
    <section className="List">
      <header className="List-header">
        <h2>{props.header}</h2>
      </header>
      <div className="List-cards">
        {CardComponent}
      </div>
    </section>

  )
}


export default List;
