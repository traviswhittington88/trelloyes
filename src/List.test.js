import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';
import { isTSAnyKeyword, exportAllDeclaration } from '@babel/types';

describe('List component', () => {
    //smoke tests
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List />, div);
        //clean up code
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders to the UI as expected', () => {
        const tree = renderer
            .create(<List header={'List A'} key={1} cardsArray={'a'} cardsObj={{a:'could be anything'}}/>)
            .toJSON();
            expect(tree).toMatchSnapshot();
    });








})