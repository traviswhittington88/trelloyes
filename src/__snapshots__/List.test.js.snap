// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders to the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      List A
    </h2>
  </header>
  <div
    className="List-cards"
  />
</section>
`;
